File: Ontology/DataPropertyRange.cs
namespace Ontology
{
    public class DataPropertyRange:
        DataPropertyAxiom,
        IDataPropertyRange
    {
        private readonly IDataRange _range;

        public DataPropertyRange(
            IOntology               ontology,
            IDataPropertyExpression dataPropertyExpression,
            IDataRange              range
            ) : base(
                ontology,
                dataPropertyExpression)
        {
            _range = range;
        }

        IDataRange IDataPropertyRange.Range => _range;
    }
}

File: Web/ClientApp/src/app/Ontology/DataPropertyRange.ts
import { DataPropertyAxiom } from "./DataPropertyAxiom";
import { IDataPropertyRange } from "./IDataPropertyRange";
import { IDataRange } from "./IDataRange";
import { IOntology } from "./IOntology";
import { IDataPropertyExpression } from "./IPropertyExpression";

export class DataPropertyRange
    extends DataPropertyAxiom
    implements IDataPropertyRange
{
    constructor(
        ontology                     : IOntology,
        public DataPropertyExpression: IDataPropertyExpression,
        public Range                 : IDataRange
        )
    {
        super(ontology);
    }
}
